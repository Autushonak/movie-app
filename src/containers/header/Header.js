import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { useRouteMatch } from 'react-router-dom';
import img from '../../static/header-background.jpg';
import HeaderContentSearch from '../header-content-search/HeaderContentSearch';
import HeaderContentMovie from '../header-content-movie/HeaderContentMovie';
import { Logo } from '../../components';

const HeaderStyled = styled.div`
  width: 100%;
  height: ${({ height }) => height}px;
`;

const BackgroundImage = styled.div`
  position: absolute;
  z-index: 1;
  left: 0;
  right: 0;
  background-image: url(${img});
  width: 100%;
  height: ${({ height }) => height}px;
  -webkit-filter: blur(5px);
  -moz-filter: blur(5px);
  -o-filter: blur(5px);
  -ms-filter: blur(5px);
  filter: blur(5px);
  background-size: 100% 400px;
`;

const LogoContainer = styled.div`
  position: absolute;
  top: 40px;
  left: 60px;
  z-index: 9999;
`;

const Header = () => {
  const match = useRouteMatch('/movie');
  const height = match ? 600 : 400;
  return (
    <HeaderStyled height={height}>
      <LogoContainer>
        <Logo />
      </LogoContainer>
      <BackgroundImage height={height} />
      {match ? <HeaderContentMovie height={height} /> : <HeaderContentSearch height={height} />}
    </HeaderStyled>
  );
};

Header.propTypes = {
  match: PropTypes.object,
};

export default Header;
export { HeaderStyled, BackgroundImage, LogoContainer };
