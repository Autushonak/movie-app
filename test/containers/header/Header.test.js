import 'jest-styled-components';
import React from 'react';
import * as router from 'react-router-dom';
import { shallow } from 'enzyme';
import Header, { HeaderStyled, BackgroundImage, LogoContainer } from '../../../src/containers/header/Header';

jest.mock('react-router-dom', () => ({
  useRouteMatch: jest.fn(),
}));

describe('HeaderStyled', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<HeaderStyled />);
    expect(component).toMatchSnapshot();
  });
});

describe('LogoContainer', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<LogoContainer />);
    expect(component).toMatchSnapshot();
  });
});

describe('BackgroundImage', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<BackgroundImage height={500} />);
    expect(component).toMatchSnapshot();
  });
});

describe('Header', () => {
  test('should be rendered correctly for movie page ', () => {
    router.useRouteMatch.mockImplementation(() => ({ params: { id: 'test-id' } }));
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
  test('should be rendered correctly for main page ', () => {
    router.useRouteMatch.mockImplementation(() => false);
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
