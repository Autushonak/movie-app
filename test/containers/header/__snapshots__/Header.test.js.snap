// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage should be rendered correctly 1`] = `
.c0 {
  position: absolute;
  z-index: 1;
  left: 0;
  right: 0;
  background-image: url();
  width: 100%;
  height: 500px;
  -webkit-filter: blur(5px);
  -moz-filter: blur(5px);
  -o-filter: blur(5px);
  -ms-filter: blur(5px);
  -webkit-filter: blur(5px);
  filter: blur(5px);
  background-size: 100% 400px;
}

<div
  className="c0"
  height={500}
/>
`;

exports[`Header should be rendered correctly for main page  1`] = `
<styled.div
  height={400}
>
  <styled.div>
    <Logo />
  </styled.div>
  <styled.div
    height={400}
  />
  <HeaderContentSearch
    height={400}
  />
</styled.div>
`;

exports[`Header should be rendered correctly for movie page  1`] = `
<styled.div
  height={600}
>
  <styled.div>
    <Logo />
  </styled.div>
  <styled.div
    height={600}
  />
  <HeaderContentMovie
    height={600}
  />
</styled.div>
`;

exports[`HeaderStyled should be rendered correctly 1`] = `
.c0 {
  width: 100%;
  height: px;
}

<div
  className="c0"
/>
`;

exports[`LogoContainer should be rendered correctly 1`] = `
.c0 {
  position: absolute;
  top: 40px;
  left: 60px;
  z-index: 9999;
}

<div
  className="c0"
/>
`;
